Trim README.md and link to our website more prominently.

ScottTodd opened the pull request:
* Remove "Getting Started" section entirely, since the new website covers that from a few different angles
* Remove "Architecture and Goals" section entirely. The new homepage covers these at a high level. The "Demonstrate MLIR's ability" and "Embrace standard-based ML via Vulkan" paragraphs are nice, but they need a better home :)
* Remove "Roadmap and Milestones" section entirely. It has been moved to https://google.github.io/iree/#roadmap (which then links back to GitHub... open to better ideas there)
* Reorder "Presentations and Talks"

stellaraccident commented on README.md:
```diff
@@ -1,10 +1,14 @@
+accelerators. IREE also provides flexible deployment solutions for its compiled
+ML models.
+
+See [our website](https://google.github.io/iree/) for project details, user
```
I wish this was more prominent, but I can't think of a better way to make that so right now...

ScottTodd replied:
The website address is also included off to the side: 

![image](https://user-images.githubusercontent.com/4010439/117508777-fbf5fc00-af3d-11eb-830c-7782e1d69a6e.png)

stellaraccident reviewed (commented):
I would move Presentations and Talks to the end.

ScottTodd commented:
> I would move Presentations and Talks to the end.

Under build status? Or just after communication channels? I moved it to put some space between "With that said, we welcome any kind of feedback on any communication channels!" and the section that linked to:

---

![image](https://user-images.githubusercontent.com/4010439/117507738-542bfe80-af3c-11eb-9844-827a86d2c377.png)

ScottTodd commented:
> > I would move Presentations and Talks to the end.
> 
> Under build status? Or just after communication channels? I moved it to put some space between "With that said, we welcome any kind of feedback on any communication channels!" and the section that linked to:
> 
> ![image](https://user-images.githubusercontent.com/4010439/117507738-542bfe80-af3c-11eb-9844-827a86d2c377.png)

Ok, just moved it to the end, between build status and the license.
